schema/samcli.json

```diff
@@ -143,9 +143,175 @@
+                  "description": "Environment name specifying default parameter values in the configuration file.",
+                  "default": "default"
+                },
+                "config_file": {
+                  "title": "config_file",
+                  "type": "string",
+                  "description": "Configuration file containing default parameter values.",
+                  "default": "samconfig.toml"
+                },
+                "guided": {
+                  "title": "guided",
+                  "type": "boolean",
+                  "description": "Specify this flag to allow SAM CLI to guide you through the deployment using guided prompts."
+                },
+                "template_file": {
+                  "title": "template_file",
+                  "type": "string",
+                  "description": "AWS SAM template which references built artifacts for resources in the template. (if applicable)",
+                  "default": "template.[yaml|yml|json]"
+                },
+                "no_execute_changeset": {
+                  "title": "no_execute_changeset",
+                  "type": "boolean",
+                  "description": "Indicates whether to execute the change set. Specify this flag to view stack changes before executing the change set."
+                },
+                "fail_on_empty_changeset": {
+                  "title": "fail_on_empty_changeset",
+                  "type": "boolean",
+                  "description": "Specify whether AWS SAM CLI should return a non-zero exit code if there are no changes to be made to the stack. Defaults to a non-zero exit code.",
+                  "default": true
+                },
+                "confirm_changeset": {
+                  "title": "confirm_changeset",
+                  "type": "boolean",
+                  "description": "Prompt to confirm if the computed changeset is to be deployed by SAM CLI."
+                },
+                "disable_rollback": {
+                  "title": "disable_rollback",
+                  "type": "boolean",
+                  "description": "Preserves the state of previously provisioned resources when an operation fails."
+                },
+                "on_failure": {
+                  "title": "on_failure",
+                  "type": "string",
+                  "description": "Provide an action to determine what will happen when a stack fails to create. Three actions are available:\n\n- ROLLBACK: This will rollback a stack to a previous known good state.\n\n- DELETE: The stack will rollback to a previous state if one exists, otherwise the stack will be deleted.\n\n- DO_NOTHING: The stack will not rollback or delete, this is the same as disabling rollback.\n\nDefault behaviour is ROLLBACK.\n\n\n\nThis option is mutually exclusive with --disable-rollback/--no-disable-rollback. You can provide\n--on-failure or --disable-rollback/--no-disable-rollback but not both at the same time.",
```

Comment:
do we have an idea on how this long description will look like in the customers IDEs?

Reply:
Some IDEs like VSCode seem to make smaller, scrollable windows pop up so that things can be scrolled through. This parameter specifically, in VSCode, looks like this:
![image](https://github.com/aws/aws-sam-cli/assets/51037424/56decc4e-6390-4d78-b88b-3595cae7038e)